crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1285 @@
+    ///        value.
+    ///    acquire_alignment (int): An integer value representing a time
+    ///        resolution of measure instruction starting time. Measure
+    ///        instruction should start at time which is a multiple of the
+    ///        alignment value.
+    ///    qubit_properties (list): A list of :class:`~.QubitProperties`
+    ///        objects defining the characteristics of each qubit on the
+    ///        target device. If specified the length of this list must match
+    ///        the number of qubits in the target, where the index in the list
+    ///        matches the qubit number the properties are defined for. If some
+    ///        qubits don't have properties available you can set that entry to
+    ///        ``None``
+    ///    concurrent_measurements(list): A list of sets of qubits that must be
+    ///        measured together. This must be provided
+    ///        as a nested list like ``[[0, 1], [2, 3, 4]]``.
+    ///Raises:
+    ///    ValueError: If both ``num_qubits`` and ``qubit_properties`` are both
+    ///        defined and the value of ``num_qubits`` differs from the length of
+    ///        ``qubit_properties``.
+    #[new]
+    #[pyo3(signature = (
+        description = None,
+        num_qubits = None,
+        dt = None,
+        granularity = None,
+        min_length = None,
+        pulse_alignment = None,
+        acquire_alignment = None,
+        qubit_properties = None,
+        concurrent_measurements = None,
+    ))]
+    fn new(
+        description: Option<String>,
+        num_qubits: Option<usize>,
+        dt: Option<f64>,
+        granularity: Option<i32>,
+        min_length: Option<usize>,
+        pulse_alignment: Option<i32>,
+        acquire_alignment: Option<i32>,
+        qubit_properties: Option<Vec<PyObject>>,
+        concurrent_measurements: Option<Vec<Vec<usize>>>,
+    ) -> PyResult<Self> {
+        let mut num_qubits = num_qubits;
+        if let Some(qubit_properties) = qubit_properties.as_ref() {
+            if let Some(num_qubits) = num_qubits {
+                if num_qubits != qubit_properties.len() {
+                    return Err(PyValueError::new_err(
+                        "The value of num_qubits specified does not match the \
+                            length of the input qubit_properties list",
+                    ));
+                }
+            } else {
+                num_qubits = Some(qubit_properties.len())
+            }
+        }
+        Ok(Target {
+            description,
+            num_qubits,
+            dt,
+            granularity: granularity.unwrap_or(1),
+            min_length: min_length.unwrap_or(1),
+            pulse_alignment: pulse_alignment.unwrap_or(1),
+            acquire_alignment: acquire_alignment.unwrap_or(0),
+            qubit_properties,
+            concurrent_measurements: concurrent_measurements.unwrap_or(Vec::new()),
```

Comment:
Was there any checking or semantics around the usage of this with `None`? Or is an empty list treated the same way in the python side?

Reply:
Seems to work the same with None as it does with an empty vec. I will change it to None as it was in the original `Target`.